test/integration/associations/has-many.test.js

```diff
@@ -1285,4 +1285,45 @@ describe(Support.getTestDialectTeaser('HasMany'), function() {
 
   });
 
+  describe('sourceKey with where clause in include', function() {
+    beforeEach(function() {
+      var User = this.sequelize.define('User',
+        { username: Sequelize.STRING, email: Sequelize.STRING },
```

Comment:
Could you change it to something that uses `field` (`email: { type: Sequelize.STRING, field: 'mail' }` or similar - So we can make sure this works (and continues to work) even when the field is called something other in the DB than in the model